src/core/lightning/index.ts

```diff
@@ -227,16 +237,20 @@ export const LightningMixin = (superclass) =>
         })
         if (balanceSats instanceof Error) throw balanceSats
 
+        const limitsChecker = await getLimitsChecker(this.user.id)
+        if (limitsChecker instanceof Error) throw limitsChecker
+
         // On us transaction
         if (isMyNode({ pubkey: destination }) || isPushPayment) {
           const lightningLoggerOnUs = lightningLogger.child({ onUs: true, fee: 0 })
 
-          const remainingOnUsLimit = await this.user.remainingOnUsLimit()
```

Comment:
`TODO` double-check that this logic makes sense

There seems to be a code-path where an `onUs` payment can be successfully added but then this limit check can cause the method to throw an error without the `onUs` payment being reverted. To determine if:
- this check **is not** needed here since  it's **_settled intraledger_**, or
- the check **is** needed and the transaction should be reverted because it's **_initiated via lightning_**

Reply:
@nicolasburtey do we have this in case of fraud attemps?